src/bionics.cpp

```diff
@@ -887,14 +887,7 @@ bool Character::burn_fuel( int b, bool start )
         if( !remote_fuel.empty() ) {
             fuel_available.emplace_back( remote_fuel );
             if( remote_fuel == fuel_type_sun_light ) {
-                // basic solar panel produces 50W = 1 charge/20_seconds = 180 charges/hour(3600)
-                if( is_wearing( "solarpack_on" ) ) {
-                    effective_efficiency = 0.05;
-                }
-                // quantum solar backpack = solar panel x6
-                if( is_wearing( "q_solarpack_on" ) ) {
-                    effective_efficiency = 0.3;
-                }
+                effective_efficiency = item_worn_with_flag( "SOLARPACK_ON" ).type->armor->solar_efficiency;
```

Comment:
`item_worn_with_flag` will return a null item when there is no suitable worn item. A null item has no armor data. Accessing `type->armor` will fail, as `armor` would be a null pointer. You need to check that the returned object is actually an armor item.